### DISCUSSION TITLE
Opening database or some database object should set active schema in toolbar to current database

### DISCUSSION THREAD

**Discussion by MatyCZ** [dbeaver/dbeaver#164] [Q&A] [answered]
When I need run some custom SQL script I must after opening New SQL editor set schema to current database. It would be great if dbeaver can do this automatically.

**Comment by codebling** [comment, marked as answer]
DBeaver already does this, it just doesn't work the way that you think it does. DBeaver sets the database for a new SQL Editor tab according to the database which is selected (or any of its elements is selected) in the Database Navigator/Project panel, **NOT** according to what tab has focus.

This can't be changed because it would be too disruptive to existing users, so this is unfortunately just a case of getting used to the behaviour. This issue should be closed.

**Comment by serge-rider** [comment]
Agreed.
